phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -371,6 +482,23 @@ private PhoenixResultSet executeQuery(final CompilableStatement stmt,
                                 plan =
                                         connection.getQueryServices().getOptimizer()
                                                 .optimize(PhoenixStatement.this, plan);
+                                setLastQueryPlan(plan);
+
+                                //verify metadata for the table/view/index in the query plan
+                                if (plan.getTableRef() != null && validateLastDdlTimestamp) {
+                                    try {
+                                        validateLastDDLTimestamp(plan.getTableRef(), true);
+                                    } catch (StaleMetadataCacheException e) {
+                                        String planSchemaName = plan.getTableRef().getTable().getSchemaName().toString();
+                                        String planTableName = plan.getTableRef().getTable().getTableName().toString();
+                                        // force update client metadata cache
+                                        LOGGER.debug("Force updating client metadata cache for {}", getInfoString(getLastQueryPlan().getTableRef()));
```

Comment:
If we receive StaleMetadataCacheException for any view in the view hierarchy, then we need to invlidate cache for the whole hierarchy, correct? since we don't know from the exception for which view the validation failed?

Reply:
The change in MetadataClient should take care of that as discussed here: https://github.com/apache/phoenix/pull/1666#discussion_r1342971828